packages/lexical-list/src/LexicalListNode.ts

```diff
@@ -49,6 +50,26 @@ export type SerializedListNode = Spread<
 
 export type ListType = 'number' | 'bullet' | 'check';
 
+export type ListStyleType =
+  | null // null means default
+  | string // string means custom style
```

Comment:
The only utility of having a union of string and some specific strings would be for autocomplete, but with `string` here by itself will default the autocomplete. There is a workaround though: https://www.charpeni.com/blog/open-ended-unions-and-autocomplete-with-typescript#solution
 
```suggestion
  | (string & {}) // string means custom style
```